src/script/lua_api/l_item.cpp

```diff
@@ -525,6 +525,29 @@ int ModApiItemMod::l_register_item_raw(lua_State *L)
 	return 0; /* number of results */
 }
 
+// unregister_item(name)
+int ModApiItemMod::l_unregister_item_raw(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	std::string name = luaL_checkstring(L, 1);
+
+	// Get the writable item definition manager from the server
+	IWritableItemDefManager *idef =
+			getServer(L)->getWritableItemDefManager();
+
+	// Unregister the node
+	if (idef->get(name).type == ITEM_NODE) {
+		IWritableNodeDefManager *ndef =
+			getServer(L)->getWritableNodeDefManager();
+		ndef->unallocateName(name);
+	}
+
+	// Unregister item
+	idef->unregisterItem(name);
+
+	return 0; /* number of results */
```

Comment:
Furthermore, the minetest developers are not in a position to say that my comments are useless (even if there are right), because they wrote by themselves hundred of useless comments that are still there. Just in this file, look at [this comment](https://github.com/minetest/minetest/blob/master/src/script/lua_api/l_item.cpp#L476), [this one](https://github.com/minetest/minetest/blob/master/src/script/lua_api/l_item.cpp#L498), [this one](https://github.com/minetest/minetest/blob/master/src/script/lua_api/l_item.cpp#L510), or [this one](https://github.com/minetest/minetest/blob/master/src/script/lua_api/l_item.cpp#L535). I was based on these comments to write mines.